### ISSUE TITLE
Pattern that the region based isolation checker does not understand how to check

### ISSUE THREAD

**Issue by NachoSoto** [swiftlang/swift#75103]
### Reproduction

```swift
actor Actor {
  func f() throws {
    Task {
      await g()
    }
  }

  private func g(_ n: Int = 1) async -> Int { n }
}
```

### Expected behavior

This compiled with Xcode 16 beta 2

### Environment

```
swift-driver version: 1.111.2 Apple Swift version 6.0 (swiftlang-6.0.0.5.15 clang-1600.0.22.6)
Target: arm64-apple-macosx14.0
```

**Comment by xedin** [comment]
cc @gottesmm

**Comment by gottesmm** [comment]
This doesn't reproduce on main.

**Comment by gottesmm** [comment]
This was fixed in f0fff2e5a02d8f56fc272ed0562a888d10f17f9b.
